ui/app/js/controllers/user_edit_controller.js

```diff
@@ -1,57 +1,66 @@
 /*global sweetAlert swal */
 angular.module('app').controller('UserPermissionsCtrl',
-function ($scope, $modalInstance, CSRF, Permissions, user, users) {
+function ($scope, $modalInstance, CSRF, Permissions, users, is_edit, user) {
 
   $scope.loading = true;
   $scope.users = users;
   $scope.currentItemTab = 1;
-  $scope.is_edit = true;
-  $scope.original_user = user;
-  $scope.user = angular.copy(user);
+  $scope.is_edit = is_edit;
   $scope.permission = {
     permission: '',
     options_as_json: ''
   };
   $scope.errors = {
     permissions: {}
   };
-  $scope.user.permissions = [];
-  Permissions.getUserPermissions(user.username)
-  .then(function(permissions) {
-    _.forEach(permissions, function(p) {
-      if (p.options) {
-        p.options_as_json = JSON.stringify(p['options']);
-      }
+  if($scope.is_edit){
+    $scope.original_user = user;
+    $scope.user = angular.copy(user);
+    $scope.loading = false;
+    $scope.user.permissions = [];
+    Permissions.getUserPermissions($scope.user.username)
+    .then(function(permissions) {
+      _.forEach(permissions, function(p) {
+        if (p.options) {
+          p.options_as_json = JSON.stringify(p['options']);
+        }
+      });
+  
+      $scope.user.permissions = permissions;
     });
 
-    $scope.user.permissions = permissions;
-    // console.log('$scope.user.permissions');
-    // console.dbg($scope.user.permissions);
+    $scope.user.roles = [];
+    Permissions.getUserRoles($scope.user.username)
+    .success(function(response) {
+      $scope.user.roles = response.roles;
+    })
+    .error(function(response) {
+      if (typeof response === 'object') {
+        $scope.errors = response;
+        sweetAlert(
+          "Form submission error",
+          "See fields highlighted in red.",
+          "error"
```

Comment:
This code runs before the user has made any changes to the Form, so "Form submission error" isn't the right message to show. How about "Failed to load User Roles"?

Reply:
Okay.